pkg/cgroup/cgroup_stats.go

```diff
@@ -17,84 +17,110 @@ limitations under the License.
 package cgroup
 
 import (
-	"bufio"
 	"fmt"
-	"os"
-	"path/filepath"
-	"regexp"
-	"strconv"
 	"strings"
 
+	"github.com/containerd/cgroups"
+	statsv1 "github.com/containerd/cgroups/stats/v1"
+	"github.com/containerd/cgroups/v3/cgroup2"
+	statsv2 "github.com/containerd/cgroups/v3/cgroup2/stats"
+
 	"github.com/sustainable-computing-io/kepler/pkg/config"
 )
 
-const (
-	ioStatFile = "io.stat"
-	reIOStat   = `(\d+):(\d+).rbytes=(\d+).wbytes=(\d+)` // 8:16 rbytes=58032128 wbytes=0 rios=120 wios=0 dbytes=0 dios=0
-)
+func getcGroupStat(pid int) (interface{}, error) {
```

Comment:
are we able to avoid using `interface{}` but use an `type interface` instead?
even if I don't want to do to deep optimism on cpu stack....
but an `interface{}` means we need at least one or two additional steps at CPU stack.
one for 
```
if cgroup==1
else
```
one for `(*statsv1.Metrics).CPU.Usage.Total` ?
considering that we will do this stack for each `PID` and memory,cpu,IO as PID * 3 times, some how it is wasted.

Reply:
It makes sense. I will update it to use `type cCgroupStat interface{}`